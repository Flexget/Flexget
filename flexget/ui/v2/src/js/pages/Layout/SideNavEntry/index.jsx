import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ListItemIcon } from 'material-ui/List';
import {
  SideNavIcon,
  SideNavText,
  NavItem,
} from './styles';

const SideNavEntry = ({ onClick, link, icon, label, className }) => {
  const item = (
    <NavItem className={className} onClick={onClick}>
      <ListItemIcon><SideNavIcon className={`fa fa-${icon}`} /></ListItemIcon>
      <SideNavText disableTypography inset primary={label} />
    </NavItem>
  );

  if (link) {
    return <Link to={link}>{item}</Link>;
  }

  return item;
};

SideNavEntry.propTypes = {
  onClick: PropTypes.func.isRequired,
  link: PropTypes.string,
  icon: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  className: PropTypes.string,
};

SideNavEntry.defaultProps = {
  className: '',
  link: '',
};

export default SideNavEntry;
