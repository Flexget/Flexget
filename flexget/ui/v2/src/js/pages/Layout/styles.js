import styled from 'react-emotion';
import theme from 'theme';

const HEADER_HEIGHT = 5;
const MOBILE_HEADER_HEIGHT = (HEADER_HEIGHT * 2);

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  min-height: 100vh;
`;

export const Main = styled.main`
  display: flex;
  flex-direction: row;
  padding-top: ${MOBILE_HEADER_HEIGHT}rem;
  flex: 1;
  height: 100vh;
  ${theme.breakpoints.up('sm')} {
    padding-top: ${HEADER_HEIGHT}rem;
  }
`;

export const Header = styled.header`
  display: flex;
  min-height: ${HEADER_HEIGHT}rem;
  flex-direction: column;
  z-index: 2;
  position: fixed;
  width: 100%;
  ${theme.breakpoints.up('sm')} {
    flex-direction: row;
  }
`;

export const LogoWrapper = styled.div`
  height: ${HEADER_HEIGHT}rem;
`;

export const Nav = styled.nav`
  height: ${HEADER_HEIGHT}rem;
  flex: 1;
`;

export const SideBar = styled.aside`
  overflow-y: auto;
`;

export const Content = styled.section`
  flex: 1;
  overflow-y: auto;
  padding: 1rem;
  opacity: 1;
  transition: ${theme.transitions.create(['opacity', 'margin-left'])};

  ${theme.breakpoints.down('sm')} {
    margin-left: 0;
    opacity: ${({ open }) => (open ? 0 : 1)};
    display: ${({ open }) => (open ? 'none' : 'block')};
  }
`;
