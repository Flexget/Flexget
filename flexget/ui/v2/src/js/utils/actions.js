import { LOADING_STATUS } from 'store/status/actions';

export const action = (type, payload, {
  message,
} = {}) => ({
  type,
  payload,
  error: (payload instanceof Error),
  meta: {
    message,
  },
});

export const request = (type, payload = {}) => ({
  type: LOADING_STATUS,
  payload,
  meta: { type },
});

export const requesting = type => act => (
  act.type === LOADING_STATUS &&
    act.meta.type === type
);
