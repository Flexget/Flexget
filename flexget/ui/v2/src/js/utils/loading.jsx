import Loadable from 'react-loadable';
import Loading from 'common/Loader';

export function createAsyncComponent(loader) {
  return Loadable({
    loading: Loading,
    loader,
  });
}
