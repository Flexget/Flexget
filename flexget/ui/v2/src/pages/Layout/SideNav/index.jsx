import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Collapse from 'material-ui/transitions/Collapse';
import Drawer from 'material-ui/Drawer';
import SideNavEntry from 'pages/Layout/SideNavEntry';
import {
  NestedSideNavEntry,
  DrawerInner,
  NavVersion,
  NavList,
  drawerPaper,
} from './styles';
import sideNavItems from './items';

export default class SideNav extends Component {
  static propTypes = {
    sideBarOpen: PropTypes.bool.isRequired,
    toggle: PropTypes.func.isRequired,
  };

  state = {
    open: {},
  };

  componentWillReceiveProps(nextProps) {
    if (this.props.sideBarOpen && !nextProps.sideBarOpen) {
      this.setState({ open: {} });
    }
  }

  toggleOnMobile = () => {
    if (
      window.matchMedia &&
      window.matchMedia('(max-width: 600px)').matches
    ) {
      this.props.toggle();
    }
  }

  handleOnClick = ({ link, label }) => {
    if (link) {
      return this.toggleOnMobile;
    }
    return () => {
      this.setState({
        open: {
          ...this.state.open,
          [label]: !this.state.open[label],
        },
      });

      if (!this.props.sideBarOpen) {
        this.props.toggle();
      }
    };
  }

  renderNavItems() {
    return sideNavItems.reduce((items, props) => {
      const list = [
        ...items,
        <SideNavEntry
          key={props.label}
          onClick={this.handleOnClick(props)}
          {...props}
        />,
      ];
      if (props.children) {
        list.push(
          <Collapse
            in={this.state.open[props.label]}
            transitionDuration="auto"
            unmountOnExit
            key={`${props.label}-collapse`}
          >
            {props.children.map(child => (
              <NestedSideNavEntry
                key={child.label}
                onClick={this.handleOnClick(child)}
                {...child}
              />
            ))}
          </Collapse>
        );
      }
      return list;
    }, []);
  }

  render() {
    const { sideBarOpen } = this.props;
    return (
      <Drawer
        open={sideBarOpen}
        type="permanent"
        classes={{
          paper: drawerPaper(sideBarOpen),
        }}
      >
        <DrawerInner>
          <NavList>{this.renderNavItems()}</NavList>
          <NavVersion hide={!sideBarOpen} />
        </DrawerInner>
      </Drawer>
    );
  }
}
