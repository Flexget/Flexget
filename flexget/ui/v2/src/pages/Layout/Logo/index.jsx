import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { NavLogo } from './styles';

const Logo = ({ sideBarOpen }) => (
  <Link to="/">
    <NavLogo open={sideBarOpen} />
  </Link>
);

Logo.propTypes = {
  sideBarOpen: PropTypes.bool.isRequired,
};

export default Logo;
