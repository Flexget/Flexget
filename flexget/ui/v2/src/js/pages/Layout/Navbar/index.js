import { connect } from 'react-redux';
import { LOGOUT } from 'store/auth/actions';
import { SERVER_RELOAD, SERVER_SHUTDOWN } from 'store/server/actions';
import { request } from 'utils/actions';
import Navbar from './Navbar';

function mapStateToProps({ router }) {
  return {
    pathname: router.location.pathname,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    logout: () => dispatch(request(LOGOUT)),
    reloadServer: () => dispatch(request(SERVER_RELOAD)),
    shutdownServer: () => dispatch(request(SERVER_SHUTDOWN)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Navbar);
export { Navbar };
