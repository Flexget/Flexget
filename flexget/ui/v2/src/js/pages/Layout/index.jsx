import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Logo from 'pages/Layout/Logo';
import Navbar from 'pages/Layout/Navbar';
import SideNav from 'pages/Layout/SideNav';
import LoadingBar from 'common/LoadingBar';
import ErrorStatus from 'common/ErrorStatus';
import InfoStatus from 'common/InfoStatus';
import {
  Wrapper,
  Main,
  Header,
  LogoWrapper,
  Nav,
  SideBar,
  Content,
} from './styles';

class Layout extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };

  static defaultProps = {
    loggedIn: false,
  };

  state = {
    sideBarOpen: (window.matchMedia && !!window.matchMedia('(min-width: 600px)').matches) || false,
  };

  toggleSideBar = () => {
    this.setState({
      sideBarOpen: !this.state.sideBarOpen,
    });
  }

  render() {
    const { children } = this.props;
    const { sideBarOpen } = this.state;

    return (
      <Wrapper>
        <Header>
          <LogoWrapper>
            <Logo sideBarOpen={sideBarOpen} />
          </LogoWrapper>
          <Nav>
            <Navbar toggle={this.toggleSideBar} />
            <LoadingBar />
          </Nav>
        </Header>
        <Main>
          <SideBar>
            <SideNav
              sideBarOpen={sideBarOpen}
              toggle={this.toggleSideBar}
            />
          </SideBar>
          <Content open={sideBarOpen}>
            { children }
          </Content>
          <ErrorStatus />
          <InfoStatus />
        </Main>
      </Wrapper>
    );
  }
}

export default Layout;
