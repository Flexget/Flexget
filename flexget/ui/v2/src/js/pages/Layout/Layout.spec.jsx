import React from 'react';
import renderer from 'react-test-renderer';
import Layout from 'pages/Layout';
import { themed, router, provider } from 'utils/tests';
import fetchMock from 'fetch-mock';

function renderLayout() {
  return provider(router(themed(
    <Layout>
      <div />
    </Layout>
  )), { router: { location: { } }, version: {}, status: { loading: {} } });
}
describe('components/layout', () => {
  beforeEach(() => {
    fetchMock
      .get('/api/server/version', {});
  });

  it('renders correctly', () => {
    const tree = renderer.create(renderLayout()).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
