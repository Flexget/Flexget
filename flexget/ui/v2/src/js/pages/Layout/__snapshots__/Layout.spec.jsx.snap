// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/layout renders correctly 1`] = `
.glamor-45 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  min-height: 100vh;
}

.glamor-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 5rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  z-index: 2;
  position: fixed;
  width: 100%;
}

.glamor-1 {
  height: 5rem;
}

.glamor-0 {
  background: #263238 url(test-file-stub) no-repeat center;
  background-size: 17.5rem;
  height: 100%;
  -webkit-transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms,background-size 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms,background-size 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.glamor-9 {
  height: 5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.glamor-8 {
  position: static;
}

.glamor-7 {
  background-color: #ef6c00;
  color: rgba(255,255,255,1);
  min-height: 5rem;
}

.glamor-2 {
  color: rgba(255,255,255,1);
}

.glamor-3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.glamor-44 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding-top: 10rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100vh;
}

.glamor-42 {
  overflow-y: auto;
}

.glamor-41 {
  position: relative;
  background-color: #263238;
  height: calc(100vh - 10rem);
  width: 0;
  -webkit-transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  border-right: none !important;
}

.glamor-40 {
  width: inherit;
  height: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.glamor-35 {
  width: inherit;
}

.glamor-13 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-13:hover {
  border-left: 3px solid #ff9800;
}

.glamor-11 {
  color: #b0bec5;
}

.glamor-39 {
  color: #f44336;
  display: none;
}

.glamor-36 {
  margin: 0;
}

.glamor-43 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow-y: auto;
  padding: 1rem;
  opacity: 1;
  -webkit-transition: opacity 300ms cubic-bezier(0.4,0,0.2,1) 0ms,margin-left 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: opacity 300ms cubic-bezier(0.4,0,0.2,1) 0ms,margin-left 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

@media (min-width:600px) {
  .glamor-10 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:600px) {
  .glamor-0 {
    width: 6rem;
    background-size: 25rem;
  }
}

@media (min-width:600px) {
  .glamor-44 {
    padding-top: 5rem;
  }
}

@media (min-width:600px) {
  .glamor-41 {
    height: calc(100vh - 5rem);
    width: 6rem;
  }
}

@media (max-width:599.99px) {
  .glamor-43 {
    margin-left: 0;
    opacity: 1;
    display: block;
  }
}

<div
  className="glamor-45"
>
  <header
    className="glamor-10"
  >
    <div
      className="glamor-1"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="glamor-0"
          open={false}
        />
      </a>
    </div>
    <nav
      className="glamor-9"
    >
      <header
        className="MuiPaper-root-8 MuiPaper-shadow4-14 MuiAppBar-root-1 MuiAppBar-positionFixed-2 MuiAppBar-colorPrimary-6 mui-fixed glamor-8"
      >
        <div
          className="MuiToolbar-root-35 MuiToolbar-gutters-36 glamor-7"
        >
          <button
            className="MuiButtonBase-root-46 MuiIconButton-root-37 glamor-2"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label-43"
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-48 MuiIconButton-icon-44 fa fa-bars"
              />
            </span>
            <span
              className="MuiTouchRipple-root-55"
            />
          </button>
          <h2
            className="MuiTypography-root-62 MuiTypography-title-68 MuiTypography-colorInherit-81"
          />
          <div
            className="glamor-3"
          />
          <a
            href="/config"
            onClick={[Function]}
          >
            <button
              aria-label="Config"
              className="MuiButtonBase-root-46 MuiIconButton-root-37 glamor-2"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label-43"
              >
                <span
                  aria-hidden="true"
                  className="material-icons MuiIcon-root-48 MuiIconButton-icon-44 fa fa-pencil"
                />
              </span>
              <span
                className="MuiTouchRipple-root-55"
              />
            </button>
          </a>
          <a
            href="/log"
            onClick={[Function]}
          >
            <button
              aria-label="Log"
              className="MuiButtonBase-root-46 MuiIconButton-root-37 glamor-2"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label-43"
              >
                <span
                  aria-hidden="true"
                  className="material-icons MuiIcon-root-48 MuiIconButton-icon-44 fa fa-book"
                />
              </span>
              <span
                className="MuiTouchRipple-root-55"
              />
            </button>
          </a>
          <button
            aria-label="Manage"
            className="MuiButtonBase-root-46 MuiIconButton-root-37 glamor-2"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label-43"
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-48 MuiIconButton-icon-44 fa fa-cog"
              />
            </span>
            <span
              className="MuiTouchRipple-root-55"
            />
          </button>
        </div>
      </header>
    </nav>
  </header>
  <main
    className="glamor-44"
  >
    <aside
      className="glamor-42"
    >
      <div
        className="MuiDrawer-docked-97"
      >
        <div
          className="MuiPaper-root-8 MuiPaper-shadow0-10 MuiDrawer-paper-98 glamor-41 MuiDrawer-paperAnchorDockedLeft-103"
        >
          <div
            className="glamor-40"
          >
            <ul
              className="MuiList-root-106 MuiList-padding-107 glamor-35"
            >
              <a
                href="/config"
                onClick={[Function]}
              >
                <li
                  className="MuiListItem-root-110 MuiListItem-gutters-117 MuiListItem-default-113 glamor-13"
                  disabled={false}
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-48 MuiListItemIcon-root-120 fa fa-pencil glamor-11"
                  />
                  <div
                    className="MuiListItemText-root-121 MuiListItemText-inset-122 glamor-11"
                  >
                    Config
                  </div>
                </li>
              </a>
              <a
                href="/history"
                onClick={[Function]}
              >
                <li
                  className="MuiListItem-root-110 MuiListItem-gutters-117 MuiListItem-default-113 glamor-13"
                  disabled={false}
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-48 MuiListItemIcon-root-120 fa fa-history glamor-11"
                  />
                  <div
                    className="MuiListItemText-root-121 MuiListItemText-inset-122 glamor-11"
                  >
                    History
                  </div>
                </li>
              </a>
              <a
                href="/log"
                onClick={[Function]}
              >
                <li
                  className="MuiListItem-root-110 MuiListItem-gutters-117 MuiListItem-default-113 glamor-13"
                  disabled={false}
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-48 MuiListItemIcon-root-120 fa fa-book glamor-11"
                  />
                  <div
                    className="MuiListItemText-root-121 MuiListItemText-inset-122 glamor-11"
                  >
                    Log
                  </div>
                </li>
              </a>
              <a
                href="/movies"
                onClick={[Function]}
              >
                <li
                  className="MuiListItem-root-110 MuiListItem-gutters-117 MuiListItem-default-113 glamor-13"
                  disabled={false}
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-48 MuiListItemIcon-root-120 fa fa-film glamor-11"
                  />
                  <div
                    className="MuiListItemText-root-121 MuiListItemText-inset-122 glamor-11"
                  >
                    Movies
                  </div>
                </li>
              </a>
              <a
                href="/pending"
                onClick={[Function]}
              >
                <li
                  className="MuiListItem-root-110 MuiListItem-gutters-117 MuiListItem-default-113 glamor-13"
                  disabled={false}
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-48 MuiListItemIcon-root-120 fa fa-check glamor-11"
                  />
                  <div
                    className="MuiListItemText-root-121 MuiListItemText-inset-122 glamor-11"
                  >
                    Pending
                  </div>
                </li>
              </a>
              <a
                href="/seen"
                onClick={[Function]}
              >
                <li
                  className="MuiListItem-root-110 MuiListItem-gutters-117 MuiListItem-default-113 glamor-13"
                  disabled={false}
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-48 MuiListItemIcon-root-120 fa fa-eye glamor-11"
                  />
                  <div
                    className="MuiListItemText-root-121 MuiListItemText-inset-122 glamor-11"
                  >
                    Seen
                  </div>
                </li>
              </a>
              <a
                href="/series"
                onClick={[Function]}
              >
                <li
                  className="MuiListItem-root-110 MuiListItem-gutters-117 MuiListItem-default-113 glamor-13"
                  disabled={false}
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-48 MuiListItemIcon-root-120 fa fa-tv glamor-11"
                  />
                  <div
                    className="MuiListItemText-root-121 MuiListItemText-inset-122 glamor-11"
                  >
                    Series
                  </div>
                </li>
              </a>
              <li
                className="MuiListItem-root-110 MuiListItem-gutters-117 MuiListItem-default-113 glamor-13"
                disabled={false}
                onClick={[Function]}
              >
                <span
                  aria-hidden="true"
                  className="material-icons MuiIcon-root-48 MuiListItemIcon-root-120 fa fa-tasks glamor-11"
                />
                <div
                  className="MuiListItemText-root-121 MuiListItemText-inset-122 glamor-11"
                >
                  Tasks
                </div>
              </li>
            </ul>
            <div
              className="glamor-39"
            >
              <p
                className="glamor-36"
              >
                Version Info
              </p>
              <p
                className="glamor-36"
              >
                Flexget: 
                 
                 
              </p>
              <p
                className="glamor-36"
              >
                API: 
              </p>
            </div>
          </div>
        </div>
      </div>
    </aside>
    <section
      className="glamor-43"
      open={false}
    >
      <div />
    </section>
  </main>
</div>
`;
